examples/06_deduplication.py

```diff
@@ -0,0 +1,186 @@
+# --------------------------------------------------------------------
+#
+# Below I use a heatmap to visualize the pairwise-distance between medication names.
+# A darker color means that two medication names are closer together (i.e. more similar),
+# a lighter color means a larger distance. We can see that we are dealing with three
+# clusters - the original medication names and their misspellings that cluster around them.
+
+import seaborn as sns
+
+fig, axes = plt.subplots(1, 1, figsize=(12, 12))
+sns.heatmap(
+    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=axes
+)
+
+###############################################################################
+# Clustering to suggest corrections of misspelled names
+# -----------------------------------------------------
+#
+# The number of clusters will need some adjustment depending on the data you have.
+# If no fixed number of clusters is given, `deduplicate` tries to set it automatically
+# via the `silhouette score <https://scikit-learn.org/stable/modules/clustering.html#silhouette-coefficient>`_.
+
+
+deduplicated_data = deduplicate(data)
+
+###############################################################################
+# We can visualize the distribution of categories in the deduplicated data:
+
+deduplicated_unique_examples, deduplicated_counts = np.unique(
+    deduplicated_data, return_counts=True
+)
+deduplicated_series = pd.Series(deduplicated_counts, index=deduplicated_unique_examples)
+
+deduplicated_series.plot.barh(figsize=(10, 15))
+_ = plt.xlabel("Medication name")
+_ = plt.ylabel("Counts")
+
+###############################################################################
+# In this example we can correct all spelling mistakes by using the ideal number
+# of clusters as determined by the silhouette score.
+#
+# However, often the translation/deduplication won't be perfect and will require some tweaks.
+# In this case, we can construct and update a translation table based on the data
+# returned by `deduplicate`.
+# It consists of the (potentially) misspelled category names as indices and the
+# (potentially) correct categories as values.
+
+# create a table that maps original -> corrected categories
+translation_table = pd.Series(deduplicated_data, index=data)
+
+# remove duplicates in the original data
+translation_table = translation_table[~translation_table.index.duplicated(keep="first")]
+
+###############################################################################
+# Since the number of correct spellings will likely be much smaller than the
+# number of original categories, we can print the estimated cluster and their
+# most common exemplars (the guessed correct spelling):
+
+
+def print_corrections(spell_correct):
+    correct = np.unique(spell_correct.values)
+    for c in correct:
+        print(
+            f"Guessed correct spelling: {c} for"
+            f" {spell_correct[spell_correct==c].index.values}"
+        )
```

Comment:
```suggestion
        print(
            f"Guessed correct spelling: {c!r} for"
            f" {spell_correct[spell_correct==c].index.values}"
        )
```

Reply:
Not sure why `!r` is needed - `c` will be strings -> no `repr` call required?

Reply:
Good point, the repr call is actually a way to put the value between single quotes
```
>>> print(f"Got unexpected value: {s}")
Got unexpected value: value
>>> print(f"Got unexpected value: {s!r}")
Got unexpected value: 'value'
```

Reply:
I see, thanks for the explanation, makes sense